src/shared/analytics/shared/mixpanel-data-helpers.ts

```diff
@@ -0,0 +1,143 @@
+import { isTruthy } from 'is-truthy-ts';
+import type { Account } from 'src/background/account/Account';
+import { getAddressActivity } from 'src/ui/shared/requests/useAddressActivity';
+import { INTERNAL_SYMBOL_CONTEXT } from 'src/background/Wallet/Wallet';
+import { isReadonlyContainer } from 'src/shared/types/validators';
+import { ZerionAPI } from 'src/modules/zerion-api/zerion-api';
+import { backgroundQueryClient } from 'src/modules/query-client/query-client.background';
+import { getAddressesPortfolio } from './getTotalWalletsBalance';
+
+async function getFundedWalletsCount(addresses: string[]) {
+  // TODO: cache results and periodically make new checks only for non-funded addresses
+
+  const result = await getAddressActivity(
+    { addresses },
+    { cachePolicy: 'cache-first' }
+  );
+  if (!result) {
+    return 0;
+  }
+  return Object.values(result).reduce(
+    (sum, value) => sum + (value.active ? 1 : 0),
+    0
+  );
+}
+
+async function getPortfolioStats(addresses: string[]) {
+  return Promise.allSettled([
+    getAddressesPortfolio(addresses),
+    getFundedWalletsCount(addresses),
+  ]).then(([result1, result2]) => {
+    return {
+      totalValue:
+        result1.status === 'fulfilled' ? result1.value.total_value : null,
+      fundedCount: result2.status === 'fulfilled' ? result2.value : null,
+    };
+  });
+}
+
+function splitIntoChunks<T>(arr: T[], size: number) {
+  const chunks: T[][] = [];
+  for (let i = 0; i < arr.length; i += size) {
+    chunks.push(arr.slice(i, i + size));
+  }
+  return chunks;
+}
+
+async function getWalletsMetaByChunks(addresses: string[]) {
+  const chunks = splitIntoChunks(addresses, 10);
```

Comment:
Is there a limitation for 10 identities per request?

Reply:
Yes :(